packages/protocol/contracts/L1/TaikoL1.sol

```diff
@@ -242,14 +242,14 @@ contract TaikoL1 is EssentialContract, IHeaderSync, TaikoEvents {
     function getSyncedHeader(
         uint256 number
     ) public view override returns (bytes32 header) {
-        header = state.getL2BlockHash(number);
-        require(header != 0, "L1:number");
```

Comment:
do not check 0 return value (same behavior on TaikoL2) and rely on caller to verify that.